include/util/mmap_file.hpp

```diff
@@ -15,14 +15,14 @@ namespace util
 
 namespace detail
 {
-template <typename T, typename RegionT>
-util::vector_view<T> mmapFile(const boost::filesystem::path &file, RegionT &region)
+template <typename T, typename MmapContainerT>
```

Comment:
What is a `MmapContainerT`? What are the requirements? Should we have concepts for this? (example https://gist.github.com/daniel-j-h/af3b08a2a569d91d638006cbbe3fff83)

Reply:
In this case, it's one of:

  - `boost::iostream::mapped_file`
  - `boost::iostream::mapped_file_source`

The only change made in this PR is to improve the naming of the template type - a larger refactoring is probably in order at some point, but not critical for this PR.